codegen/gateway.go

```diff
@@ -101,32 +101,46 @@ func NewClientSpec(jsonFile string, h *PackageHelper) (*ClientSpec, error) {
 		)
 	}
 
-	clientConfigGeneric := map[string]interface{}{}
-	err = json.Unmarshal(bytes, &clientConfigGeneric)
+	classConfig := map[string]interface{}{}
+	err = json.Unmarshal(bytes, &classConfig)
 	if err != nil {
 		return nil, errors.Wrapf(
 			err, "Could not parse json file %s: ", jsonFile,
 		)
 	}
 
-	clientType := clientConfigGeneric["clientType"].(string)
-	if clientType == "http" {
+	className := classConfig["name"].(string)
+	classType := classConfig["type"].(string)
+
+	clientConfig := classConfig["config"].(map[string]interface{})
+
+	if classType == "http" {
```

Comment:
Should this type be global ? aka "httpClient" or is the type per module aka "http" type for the client module.